src/panels/lovelace/common/evaluate-filter.ts

```diff
@@ -2,11 +2,24 @@ import { HassEntity } from "home-assistant-js-websocket";
 
 export const evaluateFilter = (stateObj: HassEntity, filter: any): boolean => {
   const operator = filter.operator || "==";
-  const value = filter.value ?? filter;
-  const state = filter.attribute
+  let value = filter.value ?? filter;
+  let state = filter.attribute
     ? stateObj.attributes[filter.attribute]
     : stateObj.state;
 
+  if (operator === "==" || operator === "!=") {
+    const valueIsNumeric =
+      typeof value === "number" ||
+      (value.trim() && typeof value === "string" && !isNaN(Number(value)));
+    const stateIsNumeric =
+      typeof state === "number" ||
+      (state.trim() && typeof state === "string" && !isNaN(Number(state)));
```

Comment:
```suggestion
      (typeof state === "string" && state.trim() && !isNaN(Number(state)));
```

Reply:
I just noticed that after I pushed, was rushing to fix before you caught me. 😞

Reply:
Done